### PR TITLE
aisleriot: 3.22.9 -> 3.22.11

### PR DESCRIPTION
**Pull request by dasj19**
###### Motivation for this change
Version bump. Adjusted dependencies to please meson.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+    pkgconfig
+    intltool
+    itstool
+    libxml2
+    desktop-file-utils
   ];
 
-  nativeBuildInputs = [ pkgconfig intltool itstool wrapGAppsHook libxml2 desktop-file-utils ];
-  buildInputs = [ gtk3 librsvg guile_2_0 libcanberra-gtk3 ];
+  buildInputs = [ gtk3 ninja qt5Full librsvg guile_2_0
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:27]: Why does it depend on QT?

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+  libcanberra-gtk3 appstream-glib pysolfc yelp-tools ];
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:28]: `ninja appstream-glib yelp-tools` should likely got to `nativeBuildInputs`.

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 1 above)
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:27]: guile_3_0 is apparently supported:

https://gitlab.gnome.org/GNOME/aisleriot/-/commit/4c79d5b039cc24066ed12d6b04b43954d29e9434

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
   };
 
-  configureFlags = [
-    "--with-card-theme-formats=svg"
-    "--with-platform=gtk-only" # until they remove GConf
+  nativeBuildInputs = [
+    wrapGAppsHook
+    meson
+    pkgconfig
+    intltool
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:21]: intltool not used anymore: https://gitlab.gnome.org/GNOME/aisleriot/-/issues/13

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 1 above)
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:27]: Could you make it one dep per line too, while at it?

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix [hunk 4]
```diff
@@ -26,6 +34,17 @@ stdenv.mkDerivation rec {
...
+  prePatch = ''
+    patchShebangs cards/meson_svgz.sh
+    patchShebangs data/meson_desktopfile.py
+    patchShebangs data/icons/meson_updateiconcache.py
+    patchShebangs src/lib/meson_compileschemas.py
+  '';
+
+  mesonFlags = [
+    "-Dtheme_kde=false"
+  ];
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:46]: These should probably be above `passthru`, see https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887.

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 1 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: After I disabled the themes it appears that it no longer needs QT. Removing qt as dependency. Thanks.

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 2 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: moved. thanks

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 1 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: I replaced `guile_2_0` with just `guile`

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 3 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: Removed from dependency list. Thnaks

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix [hunk 5]
```diff
@@ -1,23 +1,64 @@
...
 
-  configureFlags = [
-    "--with-card-theme-formats=svg"
-    "--with-platform=gtk-only" # until they remove GConf
+  nativeBuildInputs = [
+    wrapGAppsHook
+    meson
+    ninja
+    appstream-glib
+    pkgconfig
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/games/aisleriot/default.nix:37]: While at it, could we use `pkg-config` instead of the `pkgconfig` alias.

Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: Yes. Done.

**Comment by jtojnar** [comment]
Hmm, looks like there is ton of the following messages at the end of the build now:

```
patchelf: cannot find section '.dynstr'. The input file is most likely statically linked
```

File: pkgs/desktops/gnome-3/games/aisleriot/default.nix (hunk 1 above)
Comment (dasj19) [reply pkgs/desktops/gnome-3/games/aisleriot/default.nix]: And `guile` resolves to version 2.2 ... version 3 is not packaged for nixos

**Comment by dasj19** [comment]
@jtojnar I do not know how to fix those, if you encountered something similar before please let me know...

**Comment by jtojnar** [comment]
Looking at the derivation output, previously:

```ShellSession
$ file result/lib/aisleriot/guile/2.0/accordion.go
result/lib/aisleriot/guile/2.0/accordion.go: Guile Object, little endian, 64bit, bytecode v2.0
```

now:

```ShellSession
$ file result/bin/.sol-wrapped 
result/bin/.sol-wrapped: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/mh78fk3x12q2a77srgkzv16h0irl8r61-glibc-2.31/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, not stripped
$ file result/lib/aisleriot/guile/2.2/athena.go
result/lib/aisleriot/guile/2.2/athena.go: ELF 64-bit LSB shared object, no machine, version 1 (embedded), dynamically linked, with debug_info, not stripped
$ patchelf --print-rpath result/lib/aisleriot/guile/2.2/athena.go
patchelf: cannot find section '.dynstr'. The input file is most likely statically linked
```

**Comment by jtojnar** [comment]
Looks like the change is intentional according to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=907061:

> I don't know all the details here, but I believe that Guile uses the ELF format,  for its byte-compiled files, and they *are* ELF files, but they're not "normal" executables.  They contain Guile VM byte code, data sections, etc.

**Comment by jtojnar** [comment]
Either way this is bug in some other component, maybe we should do the same `dh_strip` does:

> * dh_strip: Skip stripping of guile-2.2 files by skipping all all files matching "/guile/.*\.go$".  Thanks to Jeremy Bicha and Rob Browning.  (Closes: #907061)

**Pull request merged**